### PR TITLE
Add struct.rs koan file

### PR DESCRIPTION
**Pull request by corincerami**

### CODE REVIEW

**Comment by corincerami** [comment]
@crazymykl rebased

File: src/koans/struct.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  assert_eq!(rust.version, 1.5);
+}
+
+// There may be cases where you want to create a new instance of a Struct
+// that is only slightly different from an existing one
+#[test]
+fn dot_merging() {
+  struct Account {
+    holder: &'static str,
+    balance: f64
```
Comment (crazymykl) [inline src/koans/struct.rs:66]: Add a third field (like account number), that's long and obnoxious, it'll make the advantages of the `.. foo` syntax more apparent.

Comment (corincerami) [inline src/koans/struct.rs:66]: Solid idea.

File: src/koans/struct.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  assert_eq!(movie.title, "Star Wars");
+  assert_eq!(movie.runtime, 121);
+}
+
+// Structs, like anything in Rust, are immutable by default.
+// If we create a mutable instance of a Struct, we can reassign its attributes.
+#[test]
+fn mutable_structs() {
+  struct Language {
+    version: f32
```
Comment (crazymykl) [inline src/koans/struct.rs]: Using floats to store version numbers is Bad News Bears. Even if it doesn't matter here, it either makes us look ignorant or encourages the dungus.

Comment (corincerami) [inline src/koans/struct.rs]: Fiiiine. It was just a late night example, but I get what you're saying.

**Comment by corincerami** [comment]
@crazymykl Your comments have been addressed.

**Pull request merged**